practice/src/js/templates/productTemplate.js

```diff
@@ -129,6 +120,21 @@ export default class ProductTemplate {
     confirmBtn.dataset.productId = productId;
   };
 
+  /**
+   * Binds event listener to toggle the delete product modal when clicking on delete product icons.
+   */
+  bindToggleModal = () => {
+    const deleteIcons = document.querySelectorAll('.delete-product-icon');
+
+    deleteIcons.forEach((icon) => {
```

Comment:
Should use event delegation for this. https://www.freecodecamp.org/news/event-delegation-javascript/

E.g
```
const parentElement = document.querySelector('#product-list');

    parentElement.addEventListener('click', (event) => {
        if (event.target.classList.contains('delete-product-icon')) {
        ```